packages/components/src/components/NewModal/NewModal.tsx

```diff
@@ -0,0 +1,208 @@
+import { ReactNode } from 'react';
+import styled from 'styled-components';
+import { useEvent } from 'react-use';
+import { borders, Elevation, mapElevationToBackground, spacingsPx } from '@trezor/theme';
+
+import { IconButton } from '../buttons/IconButton/IconButton';
+import { Text } from '../typography/Text/Text';
+import { H3 } from '../typography/Heading/Heading';
+import { ElevationContext } from '../ElevationContext/ElevationContext';
+import { useScrollShadow } from '../../utils/useScrollShadow';
+import { NewModalButton } from './NewModalButton';
+import { NewModalContext } from './NewModalContext';
+import { ModalAlignment, NewModalBackdrop } from './NewModalBackdrop';
+import { NewModalProvider } from './NewModalProvider';
+import type { NewModalVariant } from './types';
+import { Icon, IconType } from '../Icon/Icon';
+import { mapVariantToIconBackground, mapVariantToIconBorderColor } from './utils';
+import type { UISize } from '../../config/types';
+
+const NEW_MODAL_CONTENT_ID = 'modal-content';
+const MODAL_ELEVATION = 0;
+const ICON_SIZE = 40;
+
+type NewModalSize = Extract<UISize, 'large' | 'medium' | 'small'>;
+
+const getSize = (size: NewModalSize) => {
+    switch (size) {
+        case 'small':
+            return 600;
+        case 'medium':
+            return 680;
+        case 'large':
+            return 750;
+    }
+};
+
+const Container = styled.div<{ $elevation: Elevation; $size: NewModalSize }>`
+    display: flex;
+    flex-direction: column;
+    position: relative;
+    border-radius: ${borders.radii.md};
+    transition: background 0.3s;
+    max-width: 95%;
+    min-width: 305px;
+    max-height: 80vh;
+    width: ${({ $size }) => getSize($size)}px;
+    overflow: hidden;
+    background: ${mapElevationToBackground};
+    box-shadow: ${({ theme }) => theme.boxShadowElevated};
+`;
+
+const Header = styled.header`
+    display: flex;
+    align-items: center;
+    word-break: break-word;
+    padding: ${spacingsPx.md};
+    padding-bottom: 0;
+    gap: ${spacingsPx.md};
+`;
+
+const HeadingContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    flex: 1;
+`;
+
+const ScrollContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    flex: 1;
+    height: 100%;
+    overflow-y: auto;
+
+    ::-webkit-scrollbar {
+        display: none;
+    }
```

Comment:
Why are you hiding scrollbar? And why I can see it? I think I'm missing something 😅

<img width="554" alt="image" src="https://github.com/user-attachments/assets/b5ecafc2-6302-4f3b-9567-62772e4caf9f">

Reply:
This has been taken from the old modal, dunno

Reply:
🤔 I might delete it then

Reply:
WDYT?